opencraft_courseware_search_filters/exclude_capa.py

```diff
@@ -16,7 +19,7 @@ def exclude_dictionary(self, **kwargs):
         """
         Exclude CAPA results
         """
-        exclude_dictionary = super(LmsSearchFilterGenerator, self).exclude_dictionary(
+        exclude_dictionary = super(LmsSearchFilterGenerator, self).exclude_dictionary(  # pylint: disable=bad-super-call
```

Comment:
Having to disable [bad-super-call](https://docs.pylint.org/en/1.6.0/features.html#id33) looks like a bug in the previous code?

```suggestion
        exclude_dictionary = super(ExcludeCapaSearchFilterGenerator, self).exclude_dictionary(
```

Reply:
It might have been a bug, but the thing is that the `exclude_dictionary` in the `LmsSearchFilterGenerator` and in the parent (`SearchFilterGenerator`) are really different. That's why I disabled the `bad-super-call`.

### SearchFilterGenerator
```
def exclude_dictionary(self, **kwargs):
    """ base implementation which excludes nothing """
    return {}
```

Meanwhile, 

### LmsSearchFilterGenerator
```
def exclude_dictionary(self, **kwargs):
    """
        Exclude any courses defined outside the current org.
    """
    exclude_dictionary = super(LmsSearchFilterGenerator, self).exclude_dictionary(**kwargs)
    course_org_filter = configuration_helpers.get_current_site_orgs()
    # If we have a course filter we are ensuring that we only get those courses above
    if not course_org_filter:
        org_filter_out_set = configuration_helpers.get_all_orgs()
        if org_filter_out_set:
            exclude_dictionary['org'] = list(org_filter_out_set)

    return exclude_dictionary
```

Do you think Samuel's input on this could help?

Reply:
Sure.. @swalladge  Do you remember why you did the inheritance this way?

Reply:
ummm.. no idea. Did the way I call super cause it to inherit from SearchFilterGenerater.exclude_dictionary instead of LmsSearchFilterGenerator.exclude_dictionary? If so, then yep it's a bug.

Since we're definitely on python3 now, probably best to avoid the gotcha and just call `super().exclude_dictionary...`?

Reply:
I probably copy-pasted and forgot to update the class in the super() call... :facepalm:

Reply:
Thanks for the quick reply @swalladge :smile:

Reply:
@pomegranited I'll go for the `super()` as @swalladge suggested. I updated your suggestion, however, I realized that I also need to update the Test. So I'll push those as a single commit.